app/views/sales/index.html.erb

```diff
@@ -0,0 +1,212 @@
+<% if @is_vendor %>
+<div>
+  <h1><%= @vendor.name %> Sales</h1>
+  <%= link_to "Add a Sale", "/vendors/#{@vendor.id}/sales/new", class: "btn btn-default" %>
+  <div class="container">
+    <h2>All Sales</h2>
+    <table class="table table-bordered">
+      <tr >
+        <th>
+          PRODUCT NAME
+        </th>
+        <th>
+          SALE
+        </th>
+        <th>
+          DATE
+        </th>
+        <th></th>
+      </tr>
+        <% total = 0 %>
+        <% @vendor.products.each do |product| %>
+          <% product.sales.length.times do |i| %>
+          <% sale = product.sales[i] %>
+          <% if i == 0 %><tr class="table-division" > <% else %> <tr> <% end %>
+          <td>
+            <% if i == 0 %>
+            <%= product.name %>
+            <% end %>
+          </td>
+           <td>
+             <%= number_to_currency(sale.amount/100) %>
+             <% total += sale.amount %>
+           </td>
+           <td>
+             <%= sale.purchase_time.strftime("%B %d, %Y") %></li>
+           </td>
+           <td>
+             <%= button_to "Delete", "/vendors/#{@vendor.id}/products/#{product.id}", method: :delete, class: "btn btn-default", data: { confirm: "Are you sure you want to delete #{product.name}?"} %>
+           </td>
+       </tr>
+         <% end %>
+      <% end %>
+      <tr class="table-division">
+        <td>
+          Total Sales:
+        </td>
+        <td>
+          <%= number_to_currency(total/100) %>
+        </td>
+        <td>
+        </td>
+      </tr>
+    </table>
+  </div>
+
+  <h2> Sale Data for This Month</h2>
+  <div class="container">
+    <table class="table table-bordered">
+      <tr >
+        <th>
+          PRODUCT NAME
+        </th>
+        <th>
+          SALE
+        </th>
+        <th>
+          DATE
+        </th>
+        <th></th>
+      </tr>
+        <% total = 0 %>
+        <% @vendor.products.each do |product| %>
+          <% first = true %>
+          <% product.sales.each do |sale| %>
+          <% if sale.purchase_time.month == DateTime.now.month && sale.purchase_time.year == DateTime.now.year %>
```

Comment:
It seems like this is a condition that you could extract to the model to either return true or false for a given sale object